### PR TITLE
fix: Rate limit version check if the request fails and output warning if version check fails

### LINKED ISSUE #652: wasm-pack provides no output when its version check fails
## 🐛 Bug description
When wasm-pack checks to see if a more recent version is released, the error is silently swallowed. Since this is an optional behavior, it makes sense to not bother the user if they're not online or something, but (as documented in #651) -- this check *always* fails, which has been the case for a few days, and gone unnoticed.

It's unlikely that a request to crates.io will complete with a non-200 status code, it seems reasonable to at least inform the user something happened in this case.

### PR DESCRIPTION
**Pull request by drager**
This PR will fix #652 and #653.

### CODE REVIEW

File: src/build/mod.rs [hunk 1]
```diff
@@ -64,8 +64,13 @@ pub fn check_wasm_pack_versions() -> Result<WasmPackVersion, Error> {
...
             match Crate::return_wasm_pack_latest_version() {
-                Some(latest) => Ok(WasmPackVersion {local, latest}),
-                None => Ok(WasmPackVersion {local, latest: "".to_string()})
+                Ok(latest) => {
+                    match latest {
+                        Some(latest) => Ok(WasmPackVersion {local, latest}),
+                        None => Ok(WasmPackVersion {local, latest: "".to_string()})
+                    }
+                },
+                Err(err) => Err(err)
```
Comment (fitzgen) [inline src/build/mod.rs]: This error propagation could be done slightly easier and more idiomatically with `?` and then providing a default for `None` is a bit nicer with `unwrap_or_else`. Put it together and we get:

```rust
Ok(Crate::return_wasm_pack_latest_version()?.unwrap_or_else(|| WasmPackVersion { ... }))
```

File: src/manifest/mod.rs [hunk 2]
```diff
@@ -141,39 +141,45 @@ struct CrateInformation {
...
-            Some(v)
-        })
+    fn return_api_call_result(
+        current_time: DateTime<offset::Local>,
+    ) -> Result<String, failure::Error> {
+        let version = Self::return_latest_wasm_pack_version();
+
+        match version {
+            Ok(ref version) => Self::override_stamp_file(current_time, Some(&version)).ok(),
+            Err(_) => Self::override_stamp_file(current_time, None).ok(),
```
Comment (fitzgen) [inline src/manifest/mod.rs]: Probably worth a comment here explaining why we override the time stamp file even when getting the version fails.

File: src/build/mod.rs (hunk 1 above)
Comment (drager) [reply src/build/mod.rs]: Good suggestion, will do that. Thanks!

File: src/manifest/mod.rs (hunk 2 above)
Comment (drager) [reply src/manifest/mod.rs]: Absolutely!

**Comment by drager** [comment]
@fitzgen @ashleygwilliams I think this is good to go now!

**Pull request merged**